pkg/instancegroups/instancegroups.go

```diff
@@ -468,3 +468,77 @@ func (g *CloudInstanceGroup) Delete(cloud fi.Cloud) error {
 
 	return nil
 }
+
+// ScaleInstanceGroup scale the cloud resources for an InstanceGroup
+type ScaleInstanceGroup struct {
+	Cluster         *api.Cluster
+	Cloud           fi.Cloud
+	DesiredReplicas int64
+
+	Clientset simple.Clientset
+}
+
+func (c *ScaleInstanceGroup) ScaleInstanceGroup(group *api.InstanceGroup) error {
+	groups, err := FindCloudInstanceGroups(c.Cloud, c.Cluster, []*api.InstanceGroup{group}, false, nil)
+	cig := groups[group.ObjectMeta.Name]
+
+	if cig == nil {
+		return fmt.Errorf("AutoScalingGroup %q not found in cloud - skipping scaling", group.ObjectMeta.Name)
+	}
+
+	if len(groups) != 1 {
+		return fmt.Errorf("Multiple InstanceGroup resources found in cloud")
+	}
+
+	cig.asg.DesiredCapacity = &c.DesiredReplicas
+	glog.Infof("Scaling InstanceGroup %q instances size to: %v", group.ObjectMeta.Name, c.DesiredReplicas)
+
+	// decrease max cluster size
+	if c.DesiredReplicas < *cig.asg.MinSize {
```

Comment:
Ah - tricky!  I'm thinking we should just set min and max both to the desired size.  If you disagree, can you add a comment (here or in the code) to explain?